modules/core/src/utils/hi-res-timestamp.js

```diff
@@ -23,16 +23,24 @@
  */
 import {window, process, isBrowser} from './globals';
 
-// High resolution timer
-export function getTimestamp() {
-  if (!isBrowser) {
-    const secondsAndNanoseconds = process.hrtime();
-    return secondsAndNanoseconds[0] + secondsAndNanoseconds[1] / 1e6;
-  }
-  if (window.performance) {
+let hiResTimestamp;
```

Comment:
While doing the if statement outside the function is faster (in theory), this creates a "side effect" during import and defeats static tree-shaking.

Some thoughts:
* If we really want to get rid of this if statement, we could work with webpack "browser" fields etc to select version at build time.
* HOWEVER. if we are really concerned with perf, the webpack import overhead tends to add much more overhead to a function call than an if statement (if you ever trace the debugger through a call to a webpack imported function you will see what I mean). 

I'd personally just leave the if statement inside the function for now. Although admittedly, the startTimeStamp is already a side effect. We should investigate if we really need it.

Reply:
Agreed. Wasn't aware of the tree-shaking concerns. I'll clean this up tomorrow and also check if startTimeStamp is used anywhere.